src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -228,9 +230,110 @@ public static char ToCharLower(int value)
 
         public static bool TryDecodeFromUtf16(ReadOnlySpan<char> chars, Span<byte> bytes)
         {
+            Debug.Assert(chars.Length <= checked(bytes.Length * 2));
```

Comment:
Does this need to be mirrored here since its in each of the other two code paths?